cloud/amazon/ec2_asg.py

```diff
@@ -219,6 +219,25 @@ def enforce_required_arguments(module):
 
 def get_properties(autoscaling_group):
     properties = dict((attr, getattr(autoscaling_group, attr)) for attr in ASG_ATTRIBUTES)
+
+    #
+    # Ansible output is serialized to JSON but our
+    # tag list is sometimes a list of Tag-objects
+    # (as received by Boto).
+    #
+    # Since we can not easily teach python to serialize
+    # such a list we replace it with a dict-representation
+    # in those cases.
+    #
+    # Yes, this is an ugly hack. But at least it works,
+    # unlike the even uglier hack that was here previously...
+    #
+    if 'tags' in properties and properties['tags'] is list:
```

Comment:
not sure what 'is list' is, but I would expect and isinstance(properties['tags'], list)